source/material_model/rheology/drucker_prager_power.cc

```diff
@@ -0,0 +1,250 @@
+    {
+
+      template <int dim>
+      DruckerPragerPower<dim>::DruckerPragerPower ()
+        = default;
+
+
+      template <int dim>
+      const DruckerPragerParameters
+      DruckerPragerPower<dim>::compute_drucker_prager_parameters (const unsigned int composition,
+                                                                  const std::vector<double> &phase_function_values,
+                                                                  const std::vector<unsigned int> &n_phase_transitions_per_composition) const
+      {
+        DruckerPragerParameters drucker_prager_parameters;
+
+        drucker_prager_parameters.max_yield_stress = max_yield_stress;
+
+        if (phase_function_values == std::vector<double>())
+          {
+            // no phases
+            drucker_prager_parameters.angle_internal_friction = angles_internal_friction[composition];
+            drucker_prager_parameters.cohesion = cohesions[composition];
+          }
+        else
+          {
+            // Average among phases
+            drucker_prager_parameters.angle_internal_friction = MaterialModel::MaterialUtilities::phase_average_value(phase_function_values, n_phase_transitions_per_composition,
+                                                                angles_internal_friction, composition);
+            drucker_prager_parameters.cohesion = MaterialModel::MaterialUtilities::phase_average_value(phase_function_values, n_phase_transitions_per_composition,
+                                                 cohesions, composition);
+          }
+        return drucker_prager_parameters;
+      }
+
+
+      template <int dim>
+      double
+      DruckerPragerPower<dim>::compute_yield_stress (const double cohesion,
+                                                     const double angle_internal_friction,
+                                                     const double pressure,
+                                                     const double max_yield_stress) const
+      {
+        const double sin_phi = std::sin(angle_internal_friction);
+        const double cos_phi = std::cos(angle_internal_friction);
+
+        // The expression below differs from Eq. 9 of Glerum et al, 2018.
+        // There are actually three different ways of choosing this parameter, which
+        // correspond to the Drucker-Prager yield surface either
+        // circumscribing (Glerum et al 2018), middle circumscribing or
+        // inscribing the Mohr-Coulomb yield surface.
+        // See for instance Owen & Hinton, Finite Elements in Plasticity, 1980.
+        // Here the middle circumscribing approach is taken.
+        const double stress_inv_part = 1. / (std::sqrt(3.0) * (3.0 + sin_phi));
+
+        // Initial yield stress (no stabilization terms)
+        const double yield_stress = ( (dim==3)
+                                      ?
+                                      ( 6.0 * cohesion * cos_phi + 6.0 * pressure * sin_phi) * stress_inv_part
```

Comment:
can you please put () around the if and the else term?

Reply:
This was implemented in #5976, so I've made the changes there and rebased #5977 and this PR.